core/src/main/java/com/datastax/oss/driver/internal/core/graal/DefaultFeature.java

```diff
@@ -0,0 +1,122 @@
+import java.lang.reflect.Method;
+import java.util.Optional;
+import org.graalvm.nativeimage.hosted.Feature;
+import org.graalvm.nativeimage.hosted.RuntimeClassInitialization;
+import org.graalvm.nativeimage.hosted.RuntimeReflection;
+
+/**
+ * A Graal feature which determines which checks for well-known config keys corresponding to
+ * instances to be instantiated via reflection, adding each to the current reflection config.
+ */
+@AutomaticFeature
+final class DefaultFeature implements Feature {
+
+  public Config buildConfig() {
+
+    /* Copied from DefaultDriverConfigLoader, including the DEFAULT_ROOT_PATH value there as the literal
+    "datastax-java-driver" below.  Can't reference the class directly here since that will introduce
+    a requirement to specify it as a build-time dependency (as well as the slf4j logger it creates
+    via static init */
+    return ConfigFactory.defaultOverrides()
+        .withFallback(ConfigFactory.defaultApplication())
+        .withFallback(ConfigFactory.defaultReference(CqlSession.class.getClassLoader()))
+        .resolve()
+        .getConfig("datastax-java-driver");
+  }
+
+  public Optional<String> tryString(Config config, DefaultDriverOption option) {
+
+    try {
+      return Optional.of(config.getString(option.getPath()));
+    } catch (ConfigException.WrongType e) {
+      return Optional.empty();
+    } catch (ConfigException.Missing e) {
+      return Optional.empty();
+    }
+  }
+
+  public Optional<Class<?>> tryLoadClass(String configVal, String... defaultPackages) {
+
+    try {
+      if (configVal.contains(".")) {
+        return Optional.of(Class.forName(configVal));
+      } else {
+        for (String defaultPackage : defaultPackages) {
+          String qualifiedClassName = defaultPackage + "." + configVal;
+          try {
+            return Optional.of(Class.forName(qualifiedClassName));
+          } catch (ClassNotFoundException cnfe) {
+          }
+        }
+        return Optional.empty();
+      }
+    } catch (LinkageError | Exception e) {
+      return Optional.empty();
+    }
+  }
+
+  public void tryRegister(Config config, DefaultDriverOption option, String... defaultPackages) {
+
+    tryString(config, option)
+        .flatMap(configVal -> tryLoadClass(configVal, defaultPackages))
+        .ifPresent(
+            (clz) -> {
+              /* Mark the class as being initialized at build time */
+              RuntimeClassInitialization.initializeAtBuildTime(clz);
+
+              /* Register all the things as being available at runtime */
+              RuntimeReflection.register(clz);
+              for (Method m : clz.getMethods()) {
+                RuntimeReflection.register(m);
+              }
+              for (Field f : clz.getFields()) {
+                RuntimeReflection.register(f);
+              }
+              for (Constructor c : clz.getConstructors()) {
+                RuntimeReflection.register(c);
+              }
```

Comment:
Note that the inclusion of both constructors here means we don't have to worry about whether the profile or non-profile constructor will be used.  We _do_ still need to add support for profiles when determining what values to return from the config.